simde/arm/neon/cale.h

```diff
@@ -0,0 +1,165 @@
+/* SPDX-License-Identifier: MIT
+ *
+ * Permission is hereby granted, free of charge, to any person
+ * obtaining a copy of this software and associated documentation
+ * files (the "Software"), to deal in the Software without
+ * restriction, including without limitation the rights to use, copy,
+ * modify, merge, publish, distribute, sublicense, and/or sell copies
+ * of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be
+ * included in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * Copyright:
+ *   2023      Yi-Yen Chung <eric681@andestech.com> (Copyright owned by Andes Technology)
+ */
+
+#if !defined(SIMDE_ARM_NEON_CALE_H)
+#define SIMDE_ARM_NEON_CALE_H
+
+#include "cage.h"
+
+HEDLEY_DIAGNOSTIC_PUSH
+SIMDE_DISABLE_UNWANTED_DIAGNOSTICS
+SIMDE_BEGIN_DECLS_
+
+SIMDE_FUNCTION_ATTRIBUTES
+uint16_t
+simde_vcaleh_f16(simde_float16_t a, simde_float16_t b) {
+  #if defined(SIMDE_ARM_NEON_A64V8_NATIVE) && defined(SIMDE_ARM_NEON_FP16)
```

Comment:
does `SIMDE_FLOAT16_API_FLOAT16` (using `_Float16` as per the C standard) work with either GCC or clang on arm?

Reply:
Still, https://developer.arm.com/architectures/instruction-sets/intrinsics/#q=vcaleh_f16 only listed `A64` as a requirement; is the `defined(SIMDE_ARM_NEON_FP16)` check really needed?

Reply:
According to Wikipedia [[ref](https://en.wikipedia.org/wiki/AArch64#ARMv8.2-A)], the ARMv8-A series architecture supports the A64 instruction set. However, half-precision floating-point support was introduced in ARMv8.2-A. This means that ARMv8.1-A does not support the data type `__fp16`. Instead, ARMv8.1-A defines `SIMDE_ARM_NEON_A64V8_NATIVE`. Therefore, I believe it is necessary to check for the existence of `defined(SIMDE_ARM_NEON_FP16)` to determine whether the architecture supports `fp16`.